src/sqlancer/FoundBugException.java

```diff
@@ -0,0 +1,26 @@
+package sqlancer;
+
+import sqlancer.sqlite3.SQLite3GlobalState;
+
+public class FoundBugException extends RuntimeException {
+    private static final long serialVersionUID = 1L;
+    private final Reproducer reproducer;
+
+    public interface Reproducer {
+        boolean bugStillTriggers(SQLite3GlobalState globalState);
```

Comment:
Just noticed that this is SQLite-specific. Would it be possible to use generics here?

Reply:
I'm trying to replace with the generics now but I'm getting all kinds of weird errors. I'll keep working on it, but for now let's assume this is fixed and review everything else.

Reply:
Just realized a fatal problem in our design: [Throwables cannot be Generic](https://stackoverflow.com/questions/501277/why-doesnt-java-allow-generic-subclasses-of-throwable). As such, I don't think we can keep a reference to a generic Reproducer without throwing away the generic type info, which is problematic. I think we'll need to rethink how this all works if we're to make this properly generic, and probably move away from using Exceptions for finding bugs.